internal/testing/patch/patch.go

```diff
@@ -0,0 +1,37 @@
+/*Package patch provides helper functions for patching static variables in tests.
+
+Ideally we would not use exported static package-level variables, but as long as
+we have them, we need to patch them in tests.
+*/
+package patch
+
+import (
+	"github.com/infrahq/secrets"
+	"gotest.tools/v3/assert"
+
+	"github.com/infrahq/infra/internal/server/models"
+)
+
+type TestingT interface {
+	assert.TestingT
+	Helper()
+	TempDir() string
+	Cleanup(func())
+}
+
+// ModelsSymmetricKey sets model.ModelsSymmetricKey to a random key for the lifetime of the test.
+// This function modifies global state, it must not be used with t.Parallel.
+func ModelsSymmetricKey(t TestingT) {
```

Comment:
We were doing this same patching in 8 different places. Sometimes I'd be fine copying a test helper like this, but in this case most of the implementations did not use the right temp dir (`os.TempDir` is not automatically cleaned up), and most were not calling `t.Cleanup` to reset the value.

I've replaced these 8 places with calls to this function.